test/Microsoft.Azure.WebJobs.Host.FunctionalTests/TableTests.cs

```diff
@@ -403,5 +439,130 @@ private class PocoWithAllTypes
 
             public Poco PocoProperty { get; set; }
         }
+
+        /// <summary>
+        /// Binding type demonstrating how custom binding extensions can be used to bind to
+        /// arbitrary types
+        /// </summary>
+        /// <typeparam name="TEntity"></typeparam>
+        public class CustomTableBinding<TEntity>
```

Comment:
The following three types show what is involved in writing a binding extension. This test example doesn't actually perform any storage operations, it just demonstrates how you would.